contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,582 @@
+     * @param _outputToken          Address of output token
+     * @param _amountSetToken       Amount SetTokens to redeem
+     * @param _minOutputReceive     Minimum amount of output token to receive
+     * @param _componentQuotes      The encoded 0x transactions execute (components -> WETH).
+     *
+     * @return outputAmount         Amount of output tokens sent to the caller
+     */
+    function redeemExactSetForToken(
+        ISetToken _setToken,
+        IERC20 _outputToken,
+        uint256 _amountSetToken,
+        uint256 _minOutputReceive,
+        ZeroExSwapQuote[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isSetToken(_setToken)
+        isWhitelistedIssuanceModule(_issuanceModule)
+        isValidInput(_setToken, _amountSetToken, _componentQuotes)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+
+        uint256 outputAmount;
+        _redeemExactSet(_setToken, _amountSetToken, _issuanceModule);
+
+        // Liquidate components for WETH and ignore _outputQuote
+        outputAmount = _sellComponentsForOutputToken(_setToken, _amountSetToken, _componentQuotes, _outputToken, _issuanceModule);
+        require(outputAmount >= _minOutputReceive, "ExchangeIssuance: INSUFFICIENT OUTPUT AMOUNT");
+
+        // Transfer sender output token
+        _outputToken.safeTransfer(msg.sender, outputAmount);
+        // Emit event
+        emit ExchangeRedeem(msg.sender, _setToken, _outputToken, _amountSetToken, outputAmount);
+        // Return output amount
+        return outputAmount;
+    }
+
+    /**
+     * Redeems an exact amount of SetTokens for ETH.
+     * The SetToken must be approved by the sender to this contract.
+     *
+     * @param _setToken             Address of the SetToken being redeemed
+     * @param _amountSetToken       Amount SetTokens to redeem
+     * @param _minEthReceive        Minimum amount of Eth to receive
+     * @param _componentQuotes      The encoded 0x transactions execute
+     *
+     * @return outputAmount         Amount of output tokens sent to the caller
+     */
+    function redeemExactSetForETH(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        uint256 _minEthReceive,
+        ZeroExSwapQuote[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isSetToken(_setToken)
+        isWhitelistedIssuanceModule(_issuanceModule)
+        isValidInput(_setToken, _amountSetToken, _componentQuotes)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+        _redeemExactSet(_setToken, _amountSetToken, _issuanceModule);
+        uint ethAmount = _sellComponentsForOutputToken(_setToken, _amountSetToken, _componentQuotes, IERC20(WETH), _issuanceModule);
+        require(ethAmount >= _minEthReceive, "ExchangeIssuance: INSUFFICIENT WETH RECEIVED");
+
+        IWETH(WETH).withdraw(ethAmount);
+        (payable(msg.sender)).sendValue(ethAmount);
```

Comment:
nit: do you need these extra parentheses? Won't `payable(msg.sender).sendValue(ethAmount)` work?

Reply:
Correct, removed those parantheses